src/runtime_src/core/pcie/emulation/cpu_em/CMakeLists.txt

```diff
@@ -32,21 +32,25 @@ set_target_properties(xrt_swemu PROPERTIES VERSION ${XRT_VERSION_STRING}
   SOVERSION ${XRT_SOVERSION})
 
 target_link_libraries(xrt_swemu
+  PUBLIC
   ${Boost_FILESYSTEM_LIBRARY}
   ${Boost_SYSTEM_LIBRARY}
   ${PROTOBUF_LIBRARY}
   xrt_coreutil
-  common_em
   rt
   uuid
+  PRIVATE
```

Comment:
I had to make this dependency PRIVATE so that I could export xrt_swemu. Same situation for xrt_hwemu. Since they share the static dependency.

Reply:
I see.  I am not a CMake expert at all, but I am guessing this means that anyone linking with XRT::xrt_swemu will also link with all the PUBLIC libraries, but not with the PRIVATE ones?    

I wonder if the implicitly linking to all the PUBLIC ones is really what we want?   Maybe everything should be private since xrt_swemu is a fully resolved library?  Same goes for all our other libraries.  

Please help me understand.

Reply:
CMake agrees with you. We should slowly transition things to private usage. Apparently, its only necessary to use PUBLIC if we are using another library in our own header files, i.e. we inherit from a class in some other library. If that is the case we must force our consumer to link to the same dependency, because it is a part of our public API.

Per [CMAKE documentation](https://cmake.org/cmake/help/v3.0/manual/cmake-buildsystem.7.html#transitive-usage-requirements):
Generally, a dependency should be specified in a use of target_link_libraries() with the **PRIVATE** keyword if it is used by only the implementation of a library, and not in the header files. If a dependency is additionally used in the header files of a library (e.g. for class inheritance), then it should be specified as a PUBLIC dependency. A dependency which is not used by the implementation of a library, but only by its headers should be specified as an INTERFACE dependency.